File: cpp/01-commands/explain/LifeGame/Cell.cpp
#include "pch.h"
#include "Cell.h"

Cell::Cell()
    : _columnPosition(-1), _linePosition(-1)
{}

Cell::Cell(int columnPosition, int linePosition)
    : _columnPosition(columnPosition), _linePosition(linePosition)
{}

bool Cell::IsEmpty() const
{
    return (_columnPosition < 0) && (_linePosition < 0);
}

int Cell::ColumnPosition() const
{
    return _columnPosition;
}

int Cell::LinePosition() const
{
    return _linePosition;
}

File: dotnet/01-commands/explain/LifeGame/Cell.cs

namespace LifeGame
{
    internal class Cell
    {
        public int ColumnPosition { get; private set; }
        public int LinePosition { get; private set; }


        public Cell(int columnPosition, int linePosition)
        {
            ColumnPosition = columnPosition;
            LinePosition = linePosition;
        }
    }
}